source/read-write-pref.txt

```diff
@@ -0,0 +1,261 @@
+
+- :manual:`Read Preference </core/read-preference/>`
+- :manual:`Read Concern </reference/read-concern/>`
+- :manual:`Write Concern </reference/write-concern/>`
+
+.. _scala-read-write-config:
+
+Configure Read and Write Operations
+-----------------------------------
+
+You can control how the driver routes read operations by setting a read preference.
+You can also control how the driver waits for acknowledgment of read and write operations
+on a replica set by setting read and write concerns.
+
+This section shows how to configure these read and write options at the following levels:
+
+- :ref:`Client <scala-read-write-client>`: Specify options in a ``MongoClientSettings`` instance
+  or in the connection URI
+- :ref:`Transaction <scala-read-write-transaction>`: Specify options in a ``TransactionOptions``
+  instance
+- :ref:`Database <scala-read-write-database>`: Call corresponding option methods on a
+  ``MongoDatabase`` instance
+- :ref:`Collection <scala-read-write-collection>`: Call corresponding option methods on a
+  ``MongoCollection`` instance
+
+.. _scala-read-write-client:
+
+Client Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to set the read preference, read concern, and
+write concern of a ``MongoClient`` instance by passing a ``MongoClientSettings``
+instance to the constructor. The code configures the following settings:
+
+- ``secondary`` read preference: Read operations retrieve data from 
+  secondary replica set members
+- ``LOCAL`` read concern: Read operations return the instance's most recent data 
+  without guaranteeing that the data has been written to a majority of the replica
+  set members
+- ``W2`` write concern: The primary and one secondary replica set member
+  must acknowledge the write operation
+
+.. literalinclude:: /includes/read-write-pref.scala
+    :language: scala
+    :dedent:
+    :start-after: start-client-settings
+    :end-before: end-client-settings
+
+Alternatively, you can specify the read and write settings in the connection
+URI, which is passed as a parameter to the ``MongoClient`` constructor:
+
+.. literalinclude:: /includes/read-write-pref.scala
+    :language: scala
+    :dedent:
+    :start-after: start-client-settings-uri
+    :end-before: end-client-settings-uri
+
+.. _scala-read-write-transaction:
+
+Transaction Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to set the read preference, read concern, and
+write concern of a transaction by passing a ``TransactionOptions``
+instance to the ``startTransaction()`` method. Transactions run within
+sessions, or groupings of related read or write operations that you
+intend to run sequentially. Before configuring transaction options, create a
+``ClientSession`` instance to start a session.
+
+.. tip::
+
+   To learn more about sessions, see :manual:`Server Sessions </reference/server-sessions/>`
+   in the {+mdb-server+} manual.
+
+The example configures the following settings:
```

Comment:
S:

```suggestion
The following code example configures the following settings:
```

Reply:
I'll leave it as "example" since I don't like the repetition of "following" personally